disttask/framework/dispatcher/dispatcher.go

```diff
@@ -250,12 +250,12 @@ func (d *dispatcher) processErrFlow(receiveErr []error) error {
 	handle := GetTaskFlowHandle(d.task.Type)
 	if handle == nil {
 		logutil.Logger(d.logCtx).Warn("gen task flow handle failed, this type handle doesn't register")
-		return d.updateTask(proto.TaskStateReverted, nil, retrySQLTimes)
+		// state transform: pending --> running --> canceling --> failed.
+		return d.updateTask(proto.TaskStateFailed, nil, retrySQLTimes)
 	}
 	meta, err := handle.ProcessErrFlow(d.ctx, d, d.task, receiveErr)
 	if err != nil {
-		logutil.Logger(d.logCtx).Warn("handle error failed", zap.Error(err))
-		return err
+		return d.handlePlanErr(handle, err)
```

Comment:
'ProcessErrFlow 'handles rollback cases. If you set the status to failed, will there be a resource leak?